frontend/src/components/signup.tsx

```diff
@@ -1,77 +1,73 @@
-import ReactDOM from 'react-dom/client';
-import { useState } from "react"; 
-import axios from 'axios';
-import { fchmod } from 'fs';
-import './signup.css';
-import { Link } from 'react-router-dom';
+/* eslint-disable @typescript-eslint/no-misused-promises */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import * as React from 'react'
+import { useState } from 'react'
+import axios from 'axios'
+import './signup.css'
+import { Link } from 'react-router-dom'
 
-export const SignUp = () => {
+export const SignUp = (): JSX.Element => {
+  const [firstName, setFirstName] = useState<string>('')
+  const [lastName, setLastName] = useState<string>('')
+  const [email, setEmail] = useState<string>('')
+  const [password, setPassword] = useState<string>('')
+  const [confirmPassword, setConfirmPassword] = useState<string>('')
+  const [warning, setWarning] = useState<string>('')
 
-    const [firstName, setFirstName] = useState<string>('');
-    const [lastName, setLastName] = useState<string>('');
-    const [email, setEmail] = useState<string>('');
-    const [password, setPassword] = useState<string>('');
-    const [confirmPassword, setConfirmPassword] = useState<string>('');
-    const [warning, setWarning] = useState<string>(''); 
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
 
-    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    if (password === confirmPassword) {
+      setWarning('')
+      await axios.post('http://localhost:8000/signup', {
+        firstName,
+        lastName,
+        email,
+        password
+      })
 
-        e.preventDefault(); 
+        .then((res) => {
+          setWarning('Successful submission.')
+        })
 
-            if(password === confirmPassword) {
-                setWarning('');
-                await axios.post('http://localhost:8000/signup', {
-                    firstName,
-                    lastName,
-                    email,
-                    password, 
-                })
-        
-                .then((res) => {
-                    setWarning('Successful submission.');
-                })
-        
-                .catch((err) => {
-                    if(err.response.status == 409){
-                        setWarning('Existing email. Please use a different email.');
-                    }
-                    else{setWarning('Submission Error. Please try again.');}
-                })
-            }
-            else{
-                setConfirmPassword(''); 
-                setWarning('Please use a different password.')
-            }
+        .catch((err) => {
+          if (err.response.status === 409) {
+            setWarning('Existing email. Please use a different email.')
+          } else { setWarning('Submission Error. Please try again.') }
+        })
+    } else {
+      setConfirmPassword('')
+      setWarning('Please use a different password.')
```

Comment:
shouldn't the warning here be more along the lines of "your passwords do not match"